src/sinks/gcp/stackdriver_metrics.rs

```diff
@@ -0,0 +1,215 @@
+                warn!("Unsupported metric type: {:?}.", not_supported);
+                None
+            }
+        }
+    }
+
+    async fn build_request(
+        &self,
+        mut metrics: Self::Output,
+    ) -> crate::Result<hyper::Request<Vec<u8>>> {
+        let metric = metrics.pop().expect("only one metric");
+        let namespace = metric.namespace().unwrap_or("namespace");
+        let metric_type = format!(
+            "custom.googleapis.com/{}/metrics/{}",
+            namespace,
+            metric.name()
+        );
+
+        let metric_labels = metric
+            .series
+            .tags
+            .unwrap_or_default()
+            .into_iter()
+            .collect::<std::collections::HashMap<_, _>>();
+        let end_time = metric.data.timestamp.unwrap_or_else(chrono::Utc::now);
+
+        let (point_value, interval, metric_kind) = match metric.data.value {
+            MetricValue::Counter { value } => {
+                let interval = gcp::GcpInterval {
+                    start_time: Some(self.started),
+                    end_time,
+                };
+
+                (value, interval, gcp::GcpMetricKind::Cumulative)
+            }
+            MetricValue::Gauge { value } => {
+                let interval = gcp::GcpInterval {
+                    start_time: None,
+                    end_time,
+                };
+
+                (value, interval, gcp::GcpMetricKind::Gauge)
+            }
+            _ => unreachable!(),
+        };
+
+        let series = gcp::GcpSeries {
+            time_series: vec![gcp::GcpSerie {
+                metric: gcp::GcpMetric {
+                    r#type: metric_type,
+                    labels: metric_labels,
+                },
+                resource: gcp::GcpResource {
+                    r#type: self.config.resource.r#type.clone(),
+                    labels: self.config.resource.labels.clone(),
+                },
+                metric_kind,
+                value_type: gcp::GcpValueType::Int64,
+                points: vec![gcp::GcpPoint {
+                    interval,
+                    value: gcp::GcpPointValue {
+                        int64_value: Some(point_value as i64),
+                    },
+                }],
+            }],
+        };
+
+        let body = serde_json::to_vec(&series).unwrap();
+        let uri: Uri = format!(
+            "https://monitoring.googleapis.com/v3/projects/{}/timeSeries",
+            self.config.project_id
+        )
+        .as_str()
+        .parse()
+        .unwrap();
+
+        let mut request = hyper::Request::post(uri)
+            .header("content-type", "application/json")
+            .body(body)
+            .unwrap();
+
+        request.headers_mut().insert(
+            AUTHORIZATION,
+            self.token
+                .header_value()
+                .unwrap()
+                .as_str()
+                .parse::<HeaderValue>()
+                .unwrap(),
+        );
```

Comment:
Can this be done while creating the request above with another `.header(…)`?

Reply:
Done